include/envoy/http/access_log.h

```diff
@@ -114,6 +114,16 @@ class RequestInfo {
    * Set whether the request is a health check request or not.
    */
   virtual void healthCheck(bool is_hc) PURE;
+
+  /**
+   * Set the downstream address
+   */
+  virtual void setDownstreamAddress(std::string address) PURE;
```

Comment:
@mattklein123 we are now passing the requestinfo to the finalizeRequestHeaders method so that it can access to information like protocol and other data about the request. (similar to access log formatter). Down stream address is added to the request info so that request_header_formatter.cc can insert CLIENT_IP into the request to the upstream clusters.